UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneExporter.cs

```diff
@@ -115,6 +122,88 @@ public GLTFSceneExporter(Transform[] rootTransforms, RetrieveTexturePathDelegate
 			return _root;
 		}
 
+		/// <summary>
+		/// Writes a binary GLB file with filename at path.
+		/// </summary>
+		/// <param name="path">File path for saving the binary file</param>
+		/// <param name="fileName">The name of the GLTF file</param>
+		public void SaveGLB(string path, string fileName)
+		{
+			Stream binStream = new MemoryStream();
+			Stream jsonStream = new MemoryStream();
+
+			_bufferWriter = new BinaryWriter(binStream);
+
+			TextWriter jsonWriter = new StreamWriter(jsonStream, System.Text.Encoding.ASCII);
+
+			_root.Scene = ExportScene(fileName, _rootTransforms);
+
+			_buffer.ByteLength = (int)_bufferWriter.BaseStream.Length;
+
+			_root.Serialize(jsonWriter);
+
+			_bufferWriter.Flush();
+			jsonWriter.Flush();
+
+			// align to 4-byte boundary to comply with spec.
+			AlignToBoundary(jsonStream);
+			AlignToBoundary(binStream, 0x00);
+
+			int glbLength = (int)(GLTFHeaderSize + SectionHeaderSize +
+				jsonStream.Length + SectionHeaderSize + binStream.Length);
+
+			string fullPath = Path.Combine(path, Path.ChangeExtension(fileName, "glb"));
+
+
+			using (FileStream glbFile = new FileStream(fullPath, FileMode.Create))
+			{
+
+				BinaryWriter writer = new BinaryWriter(glbFile);
+
+				// write header
+				writer.Write(MagicGLTF);
+				writer.Write(Version);
+				writer.Write(glbLength);
+
+				// write JSON chunk header.
+				writer.Write((int)jsonStream.Length);
+				writer.Write(MagicJson);
+
+				jsonStream.Position = 0;
+				CopyStream(jsonStream, writer);
+
+				writer.Write((int)binStream.Length);
+				writer.Write(MagicBin);
+
+				binStream.Position = 0;
+				CopyStream(binStream, writer);
+
+				writer.Flush();
+			}
+
+			ExportImages(path);
+		}
+
+		private static void CopyStream(Stream input, BinaryWriter output)
+		{
+			byte[] buffer = new byte[8 * 1024];
+			int length;
+			while ((length = input.Read(buffer, 0, buffer.Length)) > 0)
+			{
+				output.Write(buffer, 0, length);
+			}
+		}
+
+		private void AlignToBoundary(Stream stream, byte pad = (byte)' ', int boundary = 4)
+		{
+			int offset = (int)(stream.Length % boundary);
```

Comment:
this will bloat by 4 bytes on the zero case. Better version is: 

```
public static uint CalculateAlignment(uint currentSize, uint byteAlignment)
		{
			return (currentSize + byteAlignment - 1) / byteAlignment * byteAlignment;
		}
```